### PR TITLE
issue #22868 - remove LAST of remaining distribution locking (printMulticopyDocument.cpp)

### PR DESCRIPTION
**Pull request by jcarlin**
There are 6 `printMulticopyDocument.cpp` inherited classes that set `_distributeInventory = true`, which if set will `DistributeInventory::SeriesAdjust`. I added a new method `distributeInventory` to printMulticopyDocument and then call it if `_distributeInventory`.

### CODE REVIEW

File: guiclient/printMulticopyDocument.cpp [hunk 1]
```diff
@@ -322,9 +335,21 @@ void printMulticopyDocument::sPrint()
...
     emit aboutToStart(&docinfoq);
+
+
+    // Distribute inventory detail
+    int itemlocSeries = 0;
+
+    if (_distributeInventory)
+    {
+      itemlocSeries = distributeInventory(&docinfoq);
+      if (itemlocSeries <= 0)
```
Comment (ChristopherCotnoir) [inline guiclient/printMulticopyDocument.cpp]: Raise an error here and cleanup before returning.

File: guiclient/printMulticopyDocument.cpp [hunk 2]
```diff
@@ -322,9 +335,21 @@ void printMulticopyDocument::sPrint()
...
+        return;
+    }
+
```
Comment (ChristopherCotnoir) [inline guiclient/printMulticopyDocument.cpp]: Would probably make the most sense to do the above right before emitting timeToPostOneDoc, after emitting timeToMarkOnePrinted.

File: guiclient/printMulticopyDocument.cpp (hunk 1 above)
Comment (jcarlin) [inline guiclient/printMulticopyDocument.cpp]: The error should have already been reported to the user and if the itemlocseries is <= 0, cleanup will do nothing. I will make sure all of the new `distributeInventory` functions raise errors appropriately.
Looking at `printInvoice.cpp`: 
- https://github.com/xtuple/qt-client/pull/1646/files#diff-eb1d6ba4c75cf7bba15efed3cea1addbR94
(Error reported in `distributeInventory::SeriesCreate`)
- https://github.com/xtuple/qt-client/pull/1646/files#diff-eb1d6ba4c75cf7bba15efed3cea1addbR128
(missing error here)
- https://github.com/xtuple/qt-client/pull/1646/files#diff-eb1d6ba4c75cf7bba15efed3cea1addbR140
(QMessageBox)

File: guiclient/printMulticopyDocument.cpp (hunk 2 above)
Comment (jcarlin) [inline guiclient/printMulticopyDocument.cpp]: I did it this way because the user can cancel and in that case, nothing should print.

Comment (jcarlin) [inline guiclient/printMulticopyDocument.cpp]: On that note about canceling, this `while` loop would be horrible if user changed their mind and there were many documents with distribution detail... `Cancel Cancel Cancel...` But it's always been this way and not the time to change it.

File: guiclient/printMulticopyDocument.cpp (hunk 1 above)
Comment (jcarlin) [inline guiclient/printMulticopyDocument.cpp]: After reviewing, I don't see any missing error handling or cleanup execution.

File: guiclient/printMulticopyDocument.cpp (hunk 2 above)
Comment (ChristopherCotnoir) [inline guiclient/printMulticopyDocument.cpp]: The posting option is kind of separate from the printing, isn't it? It seems to me even if the posting fails or is canceled, it should still print.

File: guiclient/printMulticopyDocument.cpp (hunk 1 above)
Comment (ChristopherCotnoir) [inline guiclient/printMulticopyDocument.cpp]: Ah, I missed the messages in SeriesCreate. OK, this is fine.

**Pull request merged**